forc-plugins/forc-client/src/op/call/mod.rs

```diff
@@ -48,224 +32,157 @@ pub struct CallResponse {
-                .await
-                .expect("Failed to build transaction");
-            let tx_hash = tx.id(chain_id);
-            let tx_status = provider
-                .send_transaction_and_await_commit(tx)
-                .await
-                .expect("Failed to send transaction");
-            (tx_status, tx_hash)
+        (Some(secret_key), false) => {
+            let wallet = WalletUnlocked::new_from_private_key(secret_key, Some(provider));
+            forc_tracing::println_warning(&format!(
+                "Using account {} derived from signing key...",
+                wallet.address().hash()
+            ));
+            Ok(wallet)
         }
-    };
-
-    let receipts = tx_status
-        .take_receipts_checked(Some(&log_decoder))
-        .expect("Failed to take receipts");
-
-    let mut receipt_parser = ReceiptParser::new(&receipts, DecoderConfig::default());
-    let result = match output {
-        cmd::call::OutputFormat::Default => {
-            let data = receipt_parser
-                .extract_contract_call_data(contract_id)
-                .expect("Failed to extract contract call data");
-            ABIDecoder::default()
-                .decode_as_debug_str(&output_param, data.as_slice())
-                .expect("Failed to decode as debug string")
+        (None, true) => {
+            let password = prompt_forc_wallet_password()?;
+            let wallet = select_local_wallet_account(&password, &provider).await?;
+            Ok(wallet)
         }
-        cmd::call::OutputFormat::Raw => {
-            let token = receipt_parser
-                .parse_call(&Bech32ContractId::from(contract_id), &output_param)
-                .expect("Failed to extract contract call data");
-            token_to_string(&token).expect("Failed to convert token to string")
+        (Some(secret_key), true) => {
+            forc_tracing::println_warning(
+                "Signing key is provided while requesting to use forc-wallet. Using signing key...",
+            );
+            let wallet = WalletUnlocked::new_from_private_key(secret_key, Some(provider));
+            Ok(wallet)
         }
-    };
+    }
+}
 
+/// Processes transaction receipts, logs, and displays transaction information
+pub(crate) fn process_transaction_output(
+    receipts: &[Receipt],
+    tx_hash: &str,
+    program_abi: &sway_core::asm_generation::ProgramABI,
+    result: String,
+    mode: &cmd::call::ExecutionMode,
+    node: &crate::NodeTarget,
+    show_receipts: bool,
+) -> Result<CallResponse> {
     // print receipts
     if show_receipts {
-        let formatted_receipts = forc_util::tx_utils::format_log_receipts(&receipts, true)?;
+        let formatted_receipts = forc_util::tx_utils::format_log_receipts(receipts, true)?;
         forc_tracing::println_label_green("receipts:", &formatted_receipts);
     }
 
     // decode logs
-    let program_abi = sway_core::asm_generation::ProgramABI::Fuel(parsed_abi);
     let logs = receipts
```

Comment:
maybe this is cleaner?

```rust
let logs = receipts
    .iter()
    .filter_map(|receipt| {
        match receipt {
            Receipt::LogData { rb, data, .. } => {
                forc_util::tx_utils::decode_log_data(&rb.to_string(), data, program_abi)
                    .ok()
                    .map(|decoded| decoded.value)
            }
            _ => None
        }
    })
    .collect::<Vec<_>>();
```

Reply:
https://github.com/FuelLabs/sway/pull/7056/commits/cca851cee9c86f7f451214378a2251b9725fcc93